lib/open_food_network/products_renderer.rb

```diff
@@ -29,17 +27,20 @@ def products_json
 
     private
 
-    def load_products
+    def shop_products
       return unless @order_cycle
-      scoper = ScopeProductToHub.new(@distributor)
 
-      OrderCycleDistributedProducts.new(@order_cycle, @distributor).
-        relation.
-        order(taxon_order).
-        each { |product| scoper.scope(product) }.
-        select do |product|
-          !product.deleted? && product.has_stock_for_distribution?(@order_cycle, @distributor)
-        end
+      @shop_products ||= begin
+        scoper = ScopeProductToHub.new(@distributor)
+
+        shop_products_service.products_relation.
```

Comment:
I find a bit confusing the fact that this first method is not in its own line like the methods below. Well, I admit it's also TOC :trollface:

Reply:
TOC? Yeah, I put the service and service method call on the same line, then additional methods that alter the service's output on separate lines. I can change it if you like.

Reply:
sorry lost in langauges, TOC == OCD. I find it simpler all methods in their line but that's just my brain.